scanner/indexer/indexer.go

```diff
@@ -291,12 +303,16 @@ func (i *localIndexer) IndexContainerImage(
 	return i.libIndex.Index(ctx, manifest)
 }
 
-func getLayerHTTPClient(ctx context.Context, imgRef name.Reference, auth authn.Authenticator, timeout time.Duration) (*http.Client, error) {
+func getLayerHTTPClient(ctx context.Context, imgRef name.Reference, auth authn.Authenticator, timeout time.Duration, insecure bool) (*http.Client, error) {
 	repo := imgRef.Context()
 	reg := repo.Registry
 	tr := remoteTransport
+	if insecure {
+		tr = insecureRemoteTransport
+	}
```

Comment:
I think the package-level variables are good, as the `DefaultTransport`s are designed for that use. One perk is that it will cache network connections for re-use

Reply:
> I think the package-level variables are good,

I am OK using it for caching connections.